interface/forms/group_attendance/functions.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+/**
+ * interface/forms/group_attendance/functions.php functions for form
+ *
+ * Contains the functions used for the group attendance form
+ *
+ * Copyright (C) 2016 Shachar Zilbershlag <shachar058@gmail.com>
+ * Copyright (C) 2016 Amiel Elboim <amielboim@gmail.com>
+ *
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @author  Shachar Zilbershlag <shachar058@gmail.com>
+ * @author  Amiel Elboim <amielboim@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+require_once(dirname(__FILE__) . "/../../../library/api.inc");
+require_once(dirname(__FILE__) . "/../../../library/forms.inc");
+/**
+ * Returns form_id of an existing attendance form for group encounter (if one already exists);
+ * @param $encounter
+ * @param $group_id
+ * @return array|null
+ */
+function get_form_id_of_existing_attendance_form($encounter, $group_id){
+    $sql = "SELECT form_id FROM forms WHERE encounter = ? AND formdir = 'group_attendance' AND therapy_group_id = ? AND deleted = 0;";
+    $result = sqlQuery($sql, array($encounter, $group_id));
+    return $result;
+}
+
+/**
+ * Inserts participant data into DB
+ * @param $form_id
+ * @param $therapy_group
+ * @param $group_encounter_data
+ * @param $appt_data
+ */
+function participant_insertions($form_id, $therapy_group, $group_encounter_data,$appt_data){
+    $patientData = $_POST['patientData'];
+    foreach ($patientData as $pid => $patient){
+
+        //Insert into therapy_groups_participants_attendance table
+        insert_into_tgpa_table($form_id, $pid, $patient);
+
+        //Check if to create appt and encounter for each patient (if has certain status and 'bottom' submit was pressed, not 'add_patient' submit).
+        $create_for_patient = if_to_create_for_patient($patient['status']);
+        if($create_for_patient){
+
+            //Create appt for each patient (if there is appointment connected to encounter)
+            if(!empty($appt_data)){
+                $pc_eid = insert_patient_appt($pid, $therapy_group, $appt_data['pc_aid'], $appt_data['pc_eventDate'], $appt_data['pc_startTime'], $patient);
+            }
+
+            //Create encounter for each patient
+            $encounter_id = insert_patient_encounter($pid, $therapy_group, $group_encounter_data['date'], $patient, $appt_data['pc_aid']);
```

Comment:
Ok did this.
In the pc_aid, is it supposed to get the provider id as a param or the name?
Also - the statuses for an appointment created for a patient as a result of attendance form has different status list than the regular patient appointments. These statuses are coming from the statuses in the attendance form itself. How do we integrate this in the tracker? The tracker seems to always relate to the regular patient appt status list.
(Notice that in the appts widget in patient file we made this type of appt unclickable so as not to run into problems with the statuses in the patient tab in add_edit_event.)